src/pieces/settings.py

```diff
@@ -154,6 +159,32 @@ def startup(cls):
 			server_startup_failed()
 			sys.exit(0) # Exit the program
 
+
+	@classmethod
+	def version_check(cls):
+		"""Check the version of the pieces os in the within range"""
+		pieces_os_version = cls.get_version()
+
+		# Parse version numbers
+		os_version_parsed = semver.VersionInfo.parse(pieces_os_version)
+		min_version_parsed = semver.VersionInfo.parse(cls.PIECES_OS_MIN_VERSION)
+		max_version_parsed = semver.VersionInfo.parse(cls.PIECES_OS_MAX_VERSION)
+
+		# Check compatibility
+		if not min_version_parsed <= os_version_parsed < max_version_parsed:
+			if os_version_parsed >= max_version_parsed:
+				print("Please update your cli-agent tool. It is not compatible with the current Pieces OS version")
+				print()
+				print("https://pypi.org/project/pieces-cli/")
+			else:
+				print("Please update your Pieces OS. It is not compatible with the current cli-agent version")
+				print()
+				print_pieces_os_link()
+			print()
+			print_version_details(pieces_os_version, __version__)
+			sys.exit(0)
```

Comment:
semver should have some functions such as 'gte', and 'lt' which will let you know if a version is greater than or less than another version.

using those would be better than string comparison

Reply:
also here to avoid nested if statements we can simply do:
```python
if semver.gte(os_version_parsed, max_version_parsed):
  # need to update cli plugin
 if semver.lt(os_version_parsed, min_version_parsed):
   # need to updated pieces os
  ```

Reply:
Hey Caleb when you compare the versions you can use the comparison operators since this is not a string this is a version object